### ISSUE TITLE
plugin was built with a different version of package internal

### ISSUE THREAD

**Issue by JCzz** [golang/vscode-go#3116]
Hi there

I am compiling golang plugin from code, and then read in the plugin again - for reading in code dynamic.

This works great from the terminal, but the problem is that when reading in the plugin using vscode debugger, the build flags are not the same as when I build the plugin.

## Example: 
1. I build using: `commands := []string{"build", "-buildmode=plugin", "-o", pluginPathSo, pluginPathGo}`
2. Reading in the plugin with `plug, err := plugin.Open(path)` I get **"plugin was built with a different version of package internal"**

## Problem
I can solve the problem using the same build flags as vscode, when building my plugin, but I dont know what build flags vscode is using?

**Comment by findleyr** [comment]
CC @hyangah @suzmue 

Can you please share your settings.json?

**Comment by hyangah** [comment]
Can you check if the environment variables and the go command shown in "Go: Locate Configured Go Tools" match the env var and go command used when you were building the plugin?

**Comment by hyangah** [comment]
Do GOROOT values match?

**Comment by JCzz** [comment]
Yes, only "-ffile-prefix-map=/var/folders/.......=/tmp/go-build" described above mismatch

**Comment by hyangah** [comment]
That is a temporary directory so I doubt that matters.
I don't know what makes the difference but you can try to pass build flags to the go command and the delve and compare what's going on. 

For go command used for building the plugin, use -x -v to see details.
For delve, add that -x -v flags to `buildFlags` https://github.com/golang/vscode-go/wiki/debugging#configuration in the launch configuration, and add fields to enable logging (https://github.com/golang/vscode-go/wiki/debugging#collect-logs). That will print build details in DEBUG OUTPUT tab.

**Comment by JCzz** [comment]
Unable to set it up correctly, can you please help - thanks.

Tried the following:
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Launch Package",
            "type": "go",
            "request": "launch",
            "mode": "auto",
            "program": "${fileDirname}",
            "showLog": true,
            "logOutput": "dap",
            "buildFlags": [
                
            ]
        },
    ]
}
```

But I dont know how to "For delve, add that -x -v flags to buildFlags"?

Thanks in advance

**Comment by jainakanksha-msft** [comment]
@JCzz @hyangah have you guys resolved the issue.
I am also facing the same problem. Go version is same for my plugin and my plugin consumer application.

Thanks in Advance

**Comment by jainakanksha-msft** [comment]
With delve -x -v flag
these are the logs
Starting: /home/akanksha/.version-fox/temp/1730658600-130656/golang/packages/bin/dlv dap --listen=127.0.0.1:40821 --log-dest=3 from /home/akanksha/Documents/projects/ABGupta/pluginTest
DAP server listening at: 127.0.0.1:40821
Type 'dlv help' for list of commands.
GOROOT: /home/akanksha/.version-fox/temp/1730658600-130656/golang
GOPATH: /home/akanksha/.version-fox/temp/1730053800-40137/golang/packages
PATH: /home/akanksha/.version-fox/temp/1730658600-130656/golang/bin:/home/akanksha/.vscode-server/cli/servers/Stable-65edc4939843c90c34d61f4ce11704f09d3e5cb6/server/bin/remote-cli:/home/akanksha/.version-fox/temp/1730658600-130656/golang/packages/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
failed to initialize plugin: error opening plugin /home/akanksha/Documents/projects/ABGupta/pluginTest/test/sample_custom_component1/sample_custom_component1.so: plugin.Open("/home/akanksha/Documents/projects/ABGupta/pluginTest/test/sample_custom_component1/sample_custom_component1"): plugin was built with a different version of package internal/abi
Process 132145 has exited with status 1
Detaching
dlv dap (131848) exited with code: 0
CC - @hyangah @findleyr

**Comment by JCzz** [comment]
> @JCzz @hyangah have you guys resolved the issue.
> 
> I am also facing the same problem. Go version is same for my plugin and my plugin consumer application.
> 
> 
> 
> Thanks in Advance

Nope, still same issue

**Comment by jainakanksha-msft** [comment]
@gopherbot @findleyr @hyangah Can you please help us here.

**Comment by firelizzard18** [comment]
This may be a delve issue, not a vscode-go issue. Are you able to successfully debug it using delve, without vscode?

**Comment by 13183338039** [comment]
When running the Go project plugin on VSCode, I also encountered an issue. I found that adding new parameters to the command solves it: -gcflags="all=-N -l" ,for example,go build -race -buildmode=plugin -gcflags="all=-N -l"  ../mrapps/wc.go

**Comment by jainakanksha-msft** [comment]
How do we handle this, this flag is only needed during vscode debug. normal piepline build and application run, we can't use this flag.
Are we making it conditional?

**Comment by firelizzard18** [comment]
If you compile your binary manually and debug that binary (instead of letting delve build the binary for you), you might be able to debug the pre-built plugin. However, you will almost certainly be missing features while debugging.

`-N` disables optimizations and `-l` disables inlining. If you debug a binary that was compiled without those flags, the debugger simply won't be able to do as much. For example, you may not be able to see the value of variables. There is no way around that.
